### DISCUSSION TITLE
Class "OpenTelemetry\Contrib\Otlp\SpanExporter" not found

### DISCUSSION THREAD

**Discussion by a-h-abid** [plunkettscott/laravel-otel#7] [Q&A] [answered]
I tried to install your package. As I went to configure `App/Providers/OtelServiceProvider::spanProcessors()` and uncommented the below section, I got the error `Class "OpenTelemetry\Contrib\Otlp\SpanExporter" not found`.

```php
return [
    // new SimpleSpanProcessor(
    //     exporter: new SpanExporter(
    //         (new OtlpHttpTransportFactory())
    //             ->create('http://127.0.0.1:4318/v1/traces', 'application/x-protobuf'),
    //     ),
    // ),
];
```

Prepared a draft repo for your quick test: https://github.com/a-h-abid/laravel-otel-draft1

Please let me know what I need to do.

**Comment by plunkettscott** [comment, marked as answer]
Hi there,

Sorry for the delayed response here. I wasn't monitoring the discussions, but I have enabled notifications now. You will need to install an appropriate exporter. This package does not include any OTLP exporters by default. To make the default stuff work, install this package:

```
composer require open-telemetry/exporter-otlp ^1.0@beta
```

This is covered in the documentation for this package which I have not yet published. Sorry about that. Let me know if this doesn't resolve your problems, though.

**Reply by a-h-abid** [reply]
Thanks a lot man... finally it worked.

![Screenshot from 2023-06-16 23-28-21](https://github.com/plunkettscott/laravel-otel/assets/2020773/90f6f3ce-8410-4f7d-bae4-9e78a1c0c8d0)

Edit: Also pushed the changes to my draft repo so others can try out easily.

**Reply by plunkettscott** [reply]
Thanks! I actually want to setup a demo repo with Laravel Sail pre-configured for a number of different services (MySQL, Meillisearch, etc.) in order to showcase all the features. I'm hoping I can find some time to get that setup before I publish the documentation.
